src/platform/api/kernelApi.ts

```diff
@@ -135,19 +135,36 @@ class JupyterKernelService implements IExportedKernelService {
             };
         }
     }
-    async startKernel(spec: KernelConnectionMetadata, notebook: NotebookDocument): Promise<IKernelConnectionInfo> {
+    async startKernel(spec: KernelConnectionMetadata, owner: Uri): Promise<IKernelConnectionInfo> {
         sendTelemetryEvent(Telemetry.JupyterKernelApiUsage, undefined, {
             extensionId: this.callingExtensionId,
             pemUsed: 'startKernel'
         });
-        return this.startOrConnect(spec, notebook);
+        return this.startOrConnect(spec, this.getAssociatedNotebook(owner, true));
```

Comment:
Today our code uses NotebookDocument as the owner for kernels.
We use notebook documents as the keys for holding references to kernels, if notebook document is closed or goes out of scope, then the kernels also get disposed/cleared out.

If we want to propagate this change all the way to IKernel, it can be done, just a little bit more work (minor refactoring). But don't see the need just yet, as all that does is introduce more nullables and more `if else`, where as adding a dummy Notebook document will be sufficient, as such dummy ones are only created from outside this extension and those callers (3rd party extensions) are responsible for managing the lifetime of the kernels.

Reply:
It seems a bit dangerous to introduce the DummyNotebook in production code (someone could try to use the stored object as a full notebook if they didn't realize it wasn't a real notebook).

If nothing else, the methods should at least all throw Not Implemented, but I'd rather just see the refactor to only use what we need (just the URI?)

Reply:
> omeone could try to use the stored object as a full notebook if they didn't realize it wasn't a real notebook

I don't see how or why we'd ever try to do that. & that's why I'm going with this simple approach.
perhaps I can do that as a debt item

> should at least all throw Not Implemented

Done